streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBVersionedStore.java

```diff
@@ -333,10 +384,34 @@ interface VersionedStoreSegment {
         long segmentIdForTimestamp(long timestamp);
     }
 
+    /**
+     * A {@link VersionedStoreClient} which additionally supports batch writes into its latest
+     * value store.
+     *
+     * @param <T> the segment type used by this client
+     */
+    interface BatchWritingVersionedStoreClient<T extends VersionedStoreSegment> extends VersionedStoreClient<T> {
```

Comment:
Why do we need this interface? Can't we just add both methods to `RocksDBVersionedStoreClient` directly?

Reply:
We could, yeah. It just means we'd need to make RocksDBVersionedStoreClient non-private and require that RocksDBVersionedStoreRestoreWriteBuffer is specifically passed an instance of RocksDBVersionedStoreClient rather than any BatchWritingVersionedStoreClient. I'll make the change.